core/publication/src/main/java/org/exoplatform/services/wcm/publication/WCMComposerImpl.java

```
@@ -218,10 +219,29 @@ public Node getContent(String workspace,
       while (nodeIterator != null && nodeIterator.hasNext()) {
         node = nodeIterator.nextNode();
         viewNode = getViewableContent(node, filters);
-        if (viewNode != null) {
+        if (viewNode != null && !nodes.contains(viewNode)) {
           nodes.add(viewNode);
         }
       }
+
+      List<Node> nodesclone = nodes.stream()
+              .collect(Collectors.toList());
+      nodes = nodes.stream().filter(nodeItem -> {
+
+        List<Node> translationNodes = null;
+        try {
+          translationNodes = getRealTranslationNodes(nodeItem);
+        } catch (Exception e) {
+          throw new RuntimeException(e.getMessage());
```

The real exception will be lost during the conversion to a runtime exception. You should at least log it.

---

Done